src/components/FloatingMenu/FloatingMenu.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import classnames from "classnames";
+import React, { ComponentPropsWithoutRef, ReactNode, forwardRef } from "react";
+import styles from "./FloatingMenu.module.css";
+import useId from "../../utils/useId";
+import { Text } from "../Typography/Text";
+
+interface Props extends ComponentPropsWithoutRef<"div"> {
+  /**
+   * The size of the menu in px.
+   */
+  // TODO: Figure out whether this component should come with a standard set of
+  // T-shirt sizes
+  size: number;
```

Comment:
Is that prop supposed to represent the `width` of the menu? If that's the case we should use `width` over `size` as the latter suggests we might have a more or less dense version of the menu

But more importantly, I don't think we should let consumers of this component define that value manually. A combination of `width: fit-content` and a `max-width` would be more appropriate in my opinion.

Probably worth to loop @americanrefugee and @janogarcia to chime in on the component behaviour